apiserver/params/model.go

```diff
@@ -158,7 +158,8 @@ type ModelInfo struct {
 
 // ModelInfoResult holds the result of a ModelInfo call.
 type ModelInfoResult struct {
-	Result *ModelInfo `json:"result,omitempty"`
+	CAASModel *CAASModelInfo `json:"caas-model,omitempty"`
+	IAASModel *ModelInfo `json:"iaas-model,omitempty"`
```

Comment:
This is a pretty nice way of doing it. I wonder if the type should live outside of `ModelInfo` and `CAASModelInfo`, otherwise we're relying on them both having the same named `type` field.

I'm not too fussed at this stage though.